lib/rack/rewindable_input.rb

```diff
@@ -21,7 +21,7 @@ def initialize(app)
       end
 
       def call(env)
-        env[RACK_INPUT] = RewindableInput.new(env[RACK_INPUT])
+        env[RACK_INPUT] = RewindableInput.new(env[RACK_INPUT]) unless env[RACK_INPUT].nil?
```

Comment:
```suggestion
        if input = env[RACK_INPUT]
          env[RACK_INPUT] = RewindableInput.new(input)
        end
```

How about avoiding extra hash lookups?